hoot-core/src/main/cpp/hoot/core/conflate/extractors/WayFeatureExtractor.cpp

```diff
@@ -35,16 +35,14 @@ using namespace std;
 namespace hoot
 {
 
-WayFeatureExtractor::WayFeatureExtractor(ValueAggregator *agg)
+WayFeatureExtractor::WayFeatureExtractor(ValueAggregatorPtr agg):
+  _agg(agg)
 {
-  if (agg != 0)
-  {
-    _agg.reset(agg);
-  }
-  else
-  {
-    _agg.reset(new MeanAggregator());
-  }
```

Comment:
Are we ever going to run into a problem were `agg` ever gets an empty `ValueAggregatorPtr`?  In the current incantation ensures that `_agg` always points to at least a `MeanAggregator` object.  Maybe something like:
```
  if (!agg)
    _agg.reset(new MeanAggregator());
```

Reply:
Fair point... let me see what I can do.